### PR TITLE
ci(owasp-dast): added workflow for DAST

### PR DESCRIPTION
**Pull request by tom-rm-meyer-ISST**
## Description

Create Owasp Zap Worklfow for DAST.

Tests two apis with api key, and overall frontend.

## Pre-review checks

Please ensure to do as many of the following checks as possible, before asking for committer review:

- [x] DEPENDENCIES are up-to-date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [x] [Copyright and license header](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02) are present on all affected files

### CODE REVIEW

**Comment by tom-rm-meyer-ISST** [comment]
Hey @RoKrish14, please review. I prepared two api tests on backend level, and the full test for the frontend. Please check. We'll need to wait for [this issue](https://github.com/eclipse-tractusx/sig-infra/issues/441#issuecomment-1948787026) to be handled to make use of the secrets.

Further Question: I remember you talking about open api. PR #251 now fixes the missing api key header config in swagger ui.

File: .github/workflows/owasp-dast.yaml
Comment (RoKrish14) [inline .github/workflows/owasp-dast.yaml:1]: LGTM 👍

**Review by RoKrish14 — APPROVED** [review]

**Review by RoKrish14 — APPROVED** [review]

**Review by RoKrish14 — CHANGES_REQUESTED** [review]
Since we dont have access to secrets,
- you can remove secrets from - `secrets.API_KEY`
- keep only - `API_KEY`

**Comment by tom-rm-meyer-ISST** [comment]
> Since we dont have access to secrets,
> 
> * you can remove secrets from - `secrets.API_KEY`
> * keep only - `API_KEY`

incorporated

**Review by RoKrish14 — APPROVED** [review]

**Review by mhellmeier — APPROVED** [review]

**Pull request merged**